test/sql/function/timestamp/test_icu_time_bucket_timestamptz.test

```diff
@@ -0,0 +1,1215 @@
+
+statement error
+select time_bucket('1 month 1 day'::interval, '2018-05-05 00:00:00+02'::timestamptz, '1 hour 30 minutes'::interval);
+
+statement error
+select time_bucket('1 month 1 day'::interval, '2018-05-05 00:00:00+02'::timestamptz, '2018-05-05 00:00:00+02'::timestamptz);
+
+statement error
+select time_bucket('3 days'::interval, '2019-05-05 00:00:00+09'::timestamptz, '2000000000 months'::interval);
+
+statement error
+select time_bucket('3 days'::interval, '2019-05-05 00:00:00+09'::timestamptz, '-2000000000 months'::interval);
+
+statement error
+select time_bucket('3 months'::interval, '2019-05-05 00:00:00+00'::timestamptz, '2000000000 months'::interval);
+
+statement error
+select time_bucket('3 months'::interval, '2019-05-05 00:00:00+00'::timestamptz, '-2000000000 months'::interval);
+
+statement error
+select time_bucket('1 microseconds'::interval, '290279-12-24 (BC) 19:59:05.224191+00'::timestamptz);
+
+query I
+select time_bucket('1 microseconds'::interval, '290279-12-24 (BC) 19:59:05.224192+00'::timestamptz);
+----
+290273-12-11 (BC) 12:06:07.224192-07:52
+
+statement error
+select time_bucket('1 microseconds'::interval, '294247-01-10 04:00:54.775+00'::timestamptz);
+
+query I
+select time_bucket('1 microseconds'::interval, '294247-01-10 04:00:54.774999+00'::timestamptz);
+----
+294247-01-09 20:00:54.774999-08
+
+query II
+with t2 as (select '2000-03-26 00:00:00'::timestamptz + (n::text || 'hours')::interval as ts from  generate_series(0, 72, 3) t1(n)) select ts, time_bucket('10 seconds'::interval, ts, 'Europe/London') from t2;
+----
+2000-03-26 00:00:00-08	2000-03-26 00:00:00-08
+2000-03-26 03:00:00-08	2000-03-26 03:00:00-08
+2000-03-26 06:00:00-08	2000-03-26 06:00:00-08
+2000-03-26 09:00:00-08	2000-03-26 09:00:00-08
+2000-03-26 12:00:00-08	2000-03-26 12:00:00-08
+2000-03-26 15:00:00-08	2000-03-26 15:00:00-08
+2000-03-26 18:00:00-08	2000-03-26 18:00:00-08
+2000-03-26 21:00:00-08	2000-03-26 21:00:00-08
+2000-03-27 00:00:00-08	2000-03-27 00:00:00-08
+2000-03-27 03:00:00-08	2000-03-27 03:00:00-08
+2000-03-27 06:00:00-08	2000-03-27 06:00:00-08
+2000-03-27 09:00:00-08	2000-03-27 09:00:00-08
+2000-03-27 12:00:00-08	2000-03-27 12:00:00-08
+2000-03-27 15:00:00-08	2000-03-27 15:00:00-08
+2000-03-27 18:00:00-08	2000-03-27 18:00:00-08
+2000-03-27 21:00:00-08	2000-03-27 21:00:00-08
+2000-03-28 00:00:00-08	2000-03-28 00:00:00-08
+2000-03-28 03:00:00-08	2000-03-28 03:00:00-08
+2000-03-28 06:00:00-08	2000-03-28 06:00:00-08
+2000-03-28 09:00:00-08	2000-03-28 09:00:00-08
+2000-03-28 12:00:00-08	2000-03-28 12:00:00-08
+2000-03-28 15:00:00-08	2000-03-28 15:00:00-08
+2000-03-28 18:00:00-08	2000-03-28 18:00:00-08
+2000-03-28 21:00:00-08	2000-03-28 21:00:00-08
+2000-03-29 00:00:00-08	2000-03-29 00:00:00-08
+
+query II
+with t2 as (select '2000-10-28 00:00:00'::timestamptz + (n::text || 'hours')::interval as ts from  generate_series(0, 72, 3) t1(n)) select ts, time_bucket('10 seconds'::interval, ts, 'Europe/London') from t2;
```

Comment:
Added DST boundary tests.

Reply:
Thanks! Please resolve the conversation.